From 6e1f3e7611187a31280fa2d1bd571e20ad067b41 Mon Sep 17 00:00:00 2001
From: Mitch Harder <mitch.harder@sabayonlinux.org>
Date: Mon, 12 Sep 2011 12:07:41 -0500
Subject: [btrfs-progs-unstable (darksatanic repo)] fix arg checking.

The argument checking code for snapshots was modified in a patch
titled:
fix incorrect argument checking for "btrfs sub snap -r"

This patch was commited by Hugo Mills <hugo@carfax.org.uk> to his
development repository for btrf-progs which is being used to
collect and test the range of btrfs-progs patches submitted to the
project.

This hotfix addresses a problem in the above referrenced patch.
---
 btrfs_cmds.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/btrfs_cmds.c b/btrfs_cmds.c
index 699dfc2..26a4e6f 100644
--- a/btrfs_cmds.c
+++ b/btrfs_cmds.c
@@ -370,11 +370,11 @@ int do_clone(int argc, char **argv)
 				"Invalid arguments for subvolume snapshot\n");
 			free(argv);
 			return 1;
 		}
 	}
-	if (argc - optind != 2) {
+	if (argc - optind != 3) {
 		fprintf(stderr, "Invalid arguments for subvolume snapshot\n");
 		free(argv);
 		return 1;
 	}
 
-- 
1.7.3.4

